taskvine/src/manager/vine_cached_name.c

```diff
@@ -261,7 +263,33 @@ char *make_mini_task_cached_name(const struct vine_file *f)
 
 	return strdup(md5_to_string(digest));
 }
+/*
+Generates a cached name based on meta data for a file object of type VINE_FILE. 
+*/
+char *vine_meta_name(const struct vine_file *f, ssize_t *totalsize ){
+
+	if(f->type != VINE_FILE) return 0;
+
+	struct stat info;
+	int64_t size;
+	int64_t number_of_files;
+
+	if(stat(f->source, &info)) return 0;
+	char *mtime = ctime(&info.st_mtime);
+
+	if(path_disk_size_info_get(f->source, &size, &number_of_files)) return 0;
```

Comment:
Last thing: don't use `path_disk_size_info_get` here because it is quite expensive for large directory trees.  (And if we cared about detecting changes, we would use CACHE_FOREVER anyway.)  Just use `info.st_size` instead.